sdk/spring/azure-spring-cloud-context/src/main/java/com/azure/spring/cloud/context/core/impl/DefaultEnvironmentProvider.java

```diff
@@ -22,7 +23,24 @@ public AzureEnvironment getEnvironment() {
         return environment;
     }
 
+    public void setEnvironment(AzureCloud cloud) {
+        this.environment = toAzureEnvironment(cloud);
+    }
+
     public void setEnvironment(AzureEnvironment environment) {
         this.environment = environment;
     }
+
+    private static AzureEnvironment toAzureEnvironment(AzureCloud cloud) {
```

Comment:
I'm really not a big fan of making this AzureEnvironment a enum, since it makes extension impossiable.